packages/veritone-react-common/src/components/SimpleSearchBar/index.js

```diff
@@ -0,0 +1,221 @@
+            <Input
+              // handle esc and enter
+              onKeyDown={onKeyDown}
+              // disable browser auto complete
+              autoComplete="false"
+              // remove material's underline
+              disableUnderline
+              placeholder={placeholder || 'Search'}
+              autoFocus={focused}
+              value={value}
+              onChange={this.onChange}
+              onFocus={this.onFocus}
+              className={cx(styles.input)}
+            />
+            <Zoom in={!!(focused && showClear && isLoading)}>
+              <div className={cx(styles.loadingIcon)}>
+                <CircularProgress size={20} />
+              </div>
+            </Zoom>
+            <Zoom in={showClear} className={cx(styles.clearButton)}>
+              <IconButton onClick={this.clear}>
+                <Clear />
+              </IconButton>
+            </Zoom>
+          </div>
+          {showExtension && (
+            <Fragment>
+              <Divider />
+              <div style={{ overflow: 'scroll' }}>
+                {autocompleteResults.map(result => {
+                  // if we have plain string array
+                  // eslint-disable-next-line lodash/prefer-lodash-typecheck
+                  if (typeof result === 'string') {
+                    return (
+                      <MenuItem
+                        key={result}
+                        // eslint-disable-next-line react/jsx-no-bind
+                        onClick={() => clear() || result.onClick(result)}
+                      >
+                        {result}
+                      </MenuItem>
+                    );
+                  }
+                  // if we have data groups
+                  else {
+                    if (result.data && result.data.length) {
+                      // get the template
+                      let TemplateComponent = result.template;
+                      return (
+                        <Fragment key={Math.random()}>
+                          {// map data to template
+                          result.data.map(data => {
+                            // eslint-disable-next-line lodash/prefer-lodash-typecheck
+                            if (typeof data === 'string') {
+                              return (
+                                <MenuItem
+                                  key={data}
+                                  // eslint-disable-next-line react/jsx-no-bind
+                                  onClick={() =>
+                                    clear() || result.onClick(data)
+                                  }
+                                >
+                                  {TemplateComponent(data)}
+                                </MenuItem>
+                              );
+                            } else {
+                              return (
+                                <MenuItem
+                                  key={data.id}
```

Comment:
Is the `id` always going to be available on `data`?  Curious if there was an error when rendering in storybook since the data objects in `entities.json` don't have `id`s

Reply:
Right, data should have id. Otherwise react will throw every element should have unique key, but it will assign one, and won't cause any problem. I was going to make it required, but since different apps will be passing different data and templates here, that's not possible with prop types.